modules/common/deployment/deployment.go

```diff
@@ -38,7 +38,7 @@ func NewDeployment(
 ) *Deployment {
 	return &Deployment{
 		deployment: deployment,
-		timeout:    timeout,
+		timeout:    time.Duration(timeout) * time.Second,
```

Comment:
Is there perhaps a desire to make `timeout` a `Duration` type itself, like you've done for `WaitForDBCreatedWithTimeout`'s `requeueAfter` parameter?  I think it's fine the way you have it, but I was just curious.  Obviously changing it from `int` to `Duration` would require all the operator callers to update their use of `NewDeployment` when they update to the new `lib-common` version, so perhaps that's sub-optimal.